src/BlockEntities/DropSpenserEntity.cpp

```diff
@@ -89,14 +90,14 @@ void cDropSpenserEntity::DropSpense(cChunk & a_Chunk)
 	int SmokeDir = 0;
 	switch (Meta)
 	{
-		case E_META_DROPSPENSER_FACING_YP: SmokeDir = 4; break;  // YP & YM don't have associated smoke dirs, just do 4 (centre of block)
-		case E_META_DROPSPENSER_FACING_YM: SmokeDir = 4; break;
-		case E_META_DROPSPENSER_FACING_XM: SmokeDir = 3; break;
-		case E_META_DROPSPENSER_FACING_XP: SmokeDir = 5; break;
-		case E_META_DROPSPENSER_FACING_ZM: SmokeDir = 1; break;
-		case E_META_DROPSPENSER_FACING_ZP: SmokeDir = 7; break;
+		case E_META_DROPSPENSER_FACING_YP: SmokeDir = static_cast<int>(SmokeDirection::CENTRE); break;  // YP & YM don't have associated smoke dirs, just do 4 (centre of block)
```

Comment:
If/when this patch has merged, I'll look at the "data" portion of `BroadcastSoundParticleEffect` to see if we can remove these casts from this part of the code.